src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.OneNote/Main.cs

```diff
@@ -0,0 +1,150 @@
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+        }
+
+        /// <summary>
+        /// Return a filtered list, based on the given query
+        /// </summary>
+        /// <param name="query">The query to filter the list</param>
+        /// <returns>A filtered list, can be empty when nothing was found</returns>
+        public List<Result> Query(Query query)
+        {
+            if (!_oneNoteInstalled || query is null || string.IsNullOrWhiteSpace(query.Search))
+            {
+                return new List<Result>(0);
+            }
+
+            var pages = OneNoteProvider.FindPages(query.Search);
+
+            return pages.Select(p => new Result
+            {
+                IcoPath = _iconPath,
+                Title = p.Name,
+                QueryTextDisplay = p.Name,
+                SubTitle = @$"{p.Notebook.Name}\{p.Section.Name}",
+                Action = (_) =>
+                {
+                    p.OpenInOneNote();
+                    ShowOneNote();
+                    return true;
+                },
+                ContextData = p,
+                ToolTipData = new ToolTipData(Name, @$"{p.Notebook.Name}\{p.Section.Name}\{p.Name}"),
+            }).ToList();
+        }
+
+        /// <summary>
+        /// Return the translated plugin title.
+        /// </summary>
+        /// <returns>A translated plugin title.</returns>
+        public string GetTranslatedPluginTitle() => Resources.PluginTitle;
+
+        /// <summary>
+        /// Return the translated plugin description.
+        /// </summary>
+        /// <returns>A translated plugin description.</returns>
+        public string GetTranslatedPluginDescription() => Resources.PluginDescription;
+
+        private void OnThemeChanged(Theme currentTheme, Theme newTheme)
+        {
+            UpdateIconPath(newTheme);
+        }
+
+        [MemberNotNull(nameof(_iconPath))]
+        private void UpdateIconPath(Theme theme)
+        {
+            _iconPath = theme == Theme.Light || theme == Theme.HighContrastWhite ? "Images/oneNote.light.png" : "Images/oneNote.dark.png";
+        }
+
+        /// <summary>
+        /// Brings OneNote to the foreground and restores it if minimized.
+        /// </summary>
+        private void ShowOneNote()
+        {
+            using var process = Process.GetProcessesByName("onenote").FirstOrDefault();
+            if (process?.MainWindowHandle != null)
+            {
+                HWND handle = (HWND)process.MainWindowHandle;
```

Comment:
I think you should close the handles after zhey aren't needed anymore.

Reply:
I see no evidence that it needs to be closed, where are you seeing that? (I **am** disposing the Process itself)